src/main/java/io/cryostat/events/S3TemplateService.java

```diff
@@ -156,12 +155,12 @@ public List<Template> getTemplates() throws FlightRecorderException {
     }
 
     @Override
-    public Optional<Document> getXml(String templateName, TemplateType unused)
+    public Optional<String> getXml(String templateName, TemplateType unused)
             throws FlightRecorderException {
         try (var stream = getModel(templateName)) {
             Document doc =
```

Comment:
Can this use of `Document` be removed? At this point it's just parsing it and then serializing it again, so the only potentially useful effect is that it will perform an XML parse validation. But at this point we are reading out of our storage, so I think it would be fair enough to assume that the file was already validated before it was placed into storage, and just remove the call here.